sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,49 @@ def is_consistent(self):
+        """ list_of_derived_quantities is the input list of derived quantities in the form of a python dictionary of tuples (key, item) where key is the name of
+        the physical quantity and item is an instance of Dimension class which possesses the dimensions of the quantity. This function is designed to take in
+        a set of physical variables (all essentially dimensionful) and returns one set of all possible dimensionless quantities, as guided by the
+        Buckingham's pi theorem (a generalization of Rayleigh's method of dimensional analysis .....) Please refer and run to the unit testing file
+        /units/test/test_dimensionless.py for the format of the input .... As a heads up, we first need to define a global set of base dimensions and with
+        those we can define the derived quantities ..... """
+        exponent_matrix = []
+        # We extract the exponent matrix from the dimensional dependencies of the quantities
+        for quantity in list_of_derived_quantities:
+            element_for_quantity = [0 for k in range(len(self.base_dims))]
+            dimensions = self.get_dimensional_dependencies(list_of_derived_quantities[quantity])
+            for base_dims in dimensions:
+                element_for_quantity[self.base_dims.index(base_dims)] = dimensions[base_dims]
+            exponent_matrix.append(element_for_quantity)
+        # Next we obtain the null space of the exponent matrix
+        exponent_matrix = Matrix(exponent_matrix)
+        exponents = exponent_matrix.T.nullspace()
+        # We need to make sure that we obtain integer nullspaces because
+        # Dimension() is not able to simplify non-diophantine exponents
+        null_space_dims = len(exponents)
+        m = 1
+        while True:
+            flag = True
+            for k in range(null_space_dims):
+                for l in range(len(exponents[k])):
+                    exponents[k][l] = exponents[k][l] * m
+                    flag = flag and (exponents[k][l]-int(exponents[k][l])) == 0.0
+            if flag:
+                # Just to make sure that the exponents are all integers
+                for k in range(null_space_dims):
+                    for l in range(len(exponents[k])):
+                        exponents[k][l] = int(exponents[k][l])
+                break
+            m = m + 1
+        set_of_dimless_nums = [Dimension(1) for k in range(null_space_dims)]
+        set_of_dimless_quant = ['' for k in range(null_space_dims)]
+        # Here as a final step we take the dot product of the nullspace with the derived quantities
+        for k in range(null_space_dims):
+            m = 0
+            for quantity in list_of_derived_quantities:
+                set_of_dimless_nums[k] = set_of_dimless_nums[k]*(list_of_derived_quantities[quantity]**exponents[k][m])
+                set_of_dimless_quant[k] = set_of_dimless_quant[k] + '[' + quantity + ']' + '**(' + str(exponents[k][m]) + ')*'
```

Comment:
Why is this string necessary? Strings should go into the printing module (usually).

Reply:
Hi Francesco @Upabjojr .... thanks for your recommendations! this time I have separated the modules, one that takes care of printing the module and the other creates objects of class Dimension that represent those quantities (probably merely for testing purposes). I have also reduced the triple loop to an argument to the nullspace() function which does the job. Similar corresponding changes also reflected in the test program. Would like to know your reviews/suggestions :)

Reply:
We usually associate the printers to classes. In this case, the printer is acting on a list. It's quite unusual for SymPy. Do we really need this printer?